From eaee28bedbd53009f68fe97b5309e8c0eebe1c7c Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Wed, 30 Jul 2025 11:06:34 +0800
Subject: [PATCH 110/123] mtk: mt76: mt7996: fix testmode rx retry issue

Remove the additional mt7996_tm_update_channel call in mt7996_tm_set_rx_frames.
The second channel update also performs calibration, which can result in missing
frames transmitted from the instrument.

The original purpose of adding the additional channel update was to prevent users
from setting the TRX state before setting the channel.
However, after rebasing to single wiphy, the command "mt76-test phyX add monX" creates
a monitor interface with the parking channel already set.
Therefore, the additional channel update is no longer required.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 mt7996/testmode.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/mt7996/testmode.c b/mt7996/testmode.c
index 6da2217d..f3c0b769 100644
--- a/mt7996/testmode.c
+++ b/mt7996/testmode.c
@@ -469,9 +469,6 @@ mt7996_tm_set_rx_frames(struct mt7996_phy *phy, bool en)
 		return;
 	}
 
-	if (!td->bf_en)
-		mt7996_tm_update_channel(phy);
-
 	if (td->tx_rate_mode >= MT76_TM_TX_MODE_HE_MU) {
 		if (td->aid)
 			mt7996_tm_set(dev, SET_ID(RX_MU_AID), td->aid);
@@ -501,7 +498,7 @@ mt7996_tm_set_tx_cont(struct mt7996_phy *phy, bool en)
 	struct mt7996_dev *dev = phy->dev;
 
 	if (!en) {
-		/* trigger firmware to stop CONT TX  */
+		/* trigger firmware to stop CONT TX */
 		mt7996_tm_set(dev, SET_ID(COMMAND), RF_CMD(STOP_TEST));
 		return;
 	}
-- 
2.45.2

